ts/nni_manager/training_service/reusable/channels/heterogenousCommandChannel.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+'use strict';
+
+import { EventEmitter } from "events";
+import { delay } from "../../../common/utils";
+import { AMLEnvironmentInformation } from '../aml/amlConfig';
+import { CommandChannel, RunnerConnection } from "../commandChannel";
+import { Channel, EnvironmentInformation } from "../environment";
+import { AMLCommandChannel } from "./amlCommandChannel";
+import { WebCommandChannel } from "./webCommandChannel";
+
+class HeterogenousRunnerConnection extends RunnerConnection {
+}
+
+export class HeterogenousCommandChannel extends CommandChannel{
+    private stopping: boolean = false;
+    private amlCommandChannel: AMLCommandChannel | undefined;
```

Comment:
Consider to support it in trialDispatcher. Currently, there may be some hardcode to one platform in trial dispatcher, but it's ready to support multiple platform already. It would be a more comprehensive and clear design.

Reply:
In my point of view, trialDispatcher is used to assign trial jobs to a environment, don't need to care what kind of environment it is, or it need to maintain more code to handle environment specific logic. In current NNI design, the environment is a singleton instance, trialDispatcher use `const environmentService = component.get<EnvironmentService>(EnvironmentService);` to get the initialized environment. If we do not want to have big change, keep a single `heterogeneousEnvironmentService` to maintain all other environments will has less change. trialDispatcher just request a environment from `heterogeneousEnvironmentService` and does not need to know which kind of environment instance it is, it is transparent to trialDispatcher.

Reply:
It's time to support multiple environment service in trial dispatcher. There is no limitation or design consideration to prevent it. The change won't be too much, and that will give gpu scheduler a full picture of all resources, instead of two level scheduling.

Reply:
removed heterogeneousCommandChannel.ts file here.